### PR TITLE
fix: historical reward pricing

### PR DESCRIPTION
**Pull request by james-a-morris**
This PR aligns the FE rebate estimations with actual computation done on the scraper. I.e. we're going to be grabbing yesterday's pricing data

### CODE REVIEW

File: api/coingecko.ts [hunk 1]
```diff
@@ -107,22 +108,36 @@ const handler = async (
...
       balancerV2PoolTokens.includes(ethers.utils.getAddress(l1Token))
     ) {
+      if (dateStr) {
+        throw new InputError(
+          "Historical price not supported for BalancerV2 tokens"
+        );
+      }
       if (baseCurrency === "usd") {
-        price = await getBalancerV2TokenPrice(l1Token);
+        price = await getBalancerV2TokenPrice(l1Token, undefined);
```
Comment (dohaki) [inline api/coingecko.ts:117]: nit:
```suggestion
        price = await getBalancerV2TokenPrice(l1Token);
```

File: src/views/Bridge/hooks/useEstimatedRewards.ts [hunk 2]
```diff
@@ -37,10 +37,33 @@ export function useEstimatedRewards(
...
+  // Check if the current time is before 03:00 UTC
+  if (now.getUTCHours() < 3) {
+    currentUTCDate.setDate(currentUTCDate.getDate() - 2); // Go back two days
+  } else {
+    currentUTCDate.setDate(currentUTCDate.getDate() - 1); // Go back one day
+  }
+  const day = currentUTCDate.getUTCDate().toString().padStart(2, "0");
+  const month = (currentUTCDate.getUTCMonth() + 1).toString().padStart(2, "0");
+  const year = currentUTCDate.getUTCFullYear();
+  const yesterdaysDate = `${day}-${month}-${year}`;
```
Comment (dohaki) [inline src/views/Bridge/hooks/useEstimatedRewards.ts:57]: nit: I think we could condense this a lot more if you use the lib `luxon` which we regularly use for these kind of date handling
```ts
import { DateTime } from "luxon";

const now = DateTime.now().toUTC();
const historicPriceDate = now.minus({
  days: now.hour < 3 ? 1 : 2,
});
const historicPriceDateISO = historicPriceDate.toISODate();
```

File: src/hooks/useCoingeckoPrice.ts [hunk 3]
```diff
@@ -4,11 +4,12 @@ import getApiEndpoint from "utils/serverless-api";
 export function useCoingeckoPrice(
   l1Token: string,
   baseCurrency: string,
+  date?: string,
```
Comment (dohaki) [inline src/hooks/useCoingeckoPrice.ts:7]: nit: how about a more descriptive name like `dateISO` or at least a comment on the expected `date` format?

File: src/hooks/useTokenConversion.ts [hunk 4]
```diff
@@ -10,7 +10,11 @@ import {
 } from "utils";
 import { ConvertDecimals } from "utils/convertdecimals";
 
-export function useTokenConversion(symbol: string, baseCurrency: string) {
+export function useTokenConversion(
+  symbol: string,
+  baseCurrency: string,
+  date?: string
```
Comment (dohaki) [inline src/hooks/useTokenConversion.ts:16]: same as above

File: src/utils/serverless-api/prod/coingecko.ts [hunk 5]
```diff
@@ -6,14 +6,16 @@ export type CoingeckoApiCall = typeof coingeckoApiCall;
 
 export async function coingeckoApiCall(
   l1Token: string,
-  baseCurrency: string
+  baseCurrency: string,
+  date?: string
```
Comment (dohaki) [inline src/utils/serverless-api/prod/coingecko.ts:10]: same as above

**Review by dohaki — COMMENTED** [review]
Left a few remarks

File: src/utils/serverless-api/prod/coingecko.ts (hunk 5 above)
Comment (james-a-morris) [reply src/utils/serverless-api/prod/coingecko.ts]: https://github.com/across-protocol/frontend/pull/1118/commits/12776a2fa01a578b2b5ac9863f4b67a40941feb2

File: src/hooks/useTokenConversion.ts (hunk 4 above)
Comment (james-a-morris) [reply src/hooks/useTokenConversion.ts]: https://github.com/across-protocol/frontend/pull/1118/commits/12776a2fa01a578b2b5ac9863f4b67a40941feb2

File: src/hooks/useCoingeckoPrice.ts (hunk 3 above)
Comment (james-a-morris) [reply src/hooks/useCoingeckoPrice.ts]: https://github.com/across-protocol/frontend/pull/1118/commits/12776a2fa01a578b2b5ac9863f4b67a40941feb2

File: src/views/Bridge/hooks/useEstimatedRewards.ts (hunk 2 above)
Comment (james-a-morris) [reply src/views/Bridge/hooks/useEstimatedRewards.ts]: https://github.com/across-protocol/frontend/pull/1118/commits/0a1b763338b91ff47acac74a33ec490368fa7415

**Review by dohaki — APPROVED** [review]
Nice 👍

**Review by amateima — APPROVED** [review]
Looks good to me!

**Pull request merged**